agent/consul/config_endpoint.go

```diff
@@ -594,6 +615,21 @@ func (c *ConfigEntry) computeResolvedServiceConfig(
 			thisReply.Mode = serviceConf.Mode
 		}
 
+		if serviceConf.TLS != nil {
+			if thisReply.TLS == nil {
```

Comment:
`service-defaults` beats `proxy-defaults`, but note there's no way to easily get back to the passive (i.e. blank) default values for a specific service if these values are set for all proxies via `proxy-defaults` since `[]` or `""` in `service-defaults` will be treated as an omission and will continue to use the common default instead.